internal/services/instances.go

```diff
@@ -117,3 +111,93 @@ func (s *ServiceClient) Create(ctx context.Context, namespace, name string, cata
 		ctx, unstructureHelmChart, metav1.CreateOptions{})
 	return errors.Wrap(err, "error creating helm chart")
 }
+
+func (s *ServiceClient) List(ctx context.Context, namespace string) ([]*models.Service, error) {
+	serviceList := []*models.Service{}
+
+	unstructuredServiceList, err := s.helmChartsKubeClient.Namespace(helmchart.Namespace()).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return serviceList, nil
+		}
+		return nil, errors.Wrap(err, "fetching the service instance")
+	}
+
+	helmChartList, err := convertUnstructuredListIntoHelmCharts(unstructuredServiceList)
+	if err != nil {
+		return nil, errors.Wrap(err, "error converting unstructured list to helm charts")
+	}
+
+	filteredServiceList := filterNamespacedHelmCharts(helmChartList, namespace)
+
+	for _, srv := range filteredServiceList {
+		var catalogServicePrefix string
+		catalogServiceName := srv.GetLabels()[ServiceLabelKey]
```

Comment:
this label key is not guaranteed to exist unless you filter like in my comment above

Reply:
The service are manually filtered in the above function, so only the labeled one are there. But I agree that a filter in the get it's better. :smile: